scripts/get-pools.js

```diff
@@ -153,71 +220,121 @@ const fetchPlatform = async (token, strategy, globe) => {
+        }
+      }
+    }
+    updateProgress();
+  })());
+  await Promise.all(promises);
+  return data;
 }
 
-/* ========================================================================================================================================================================= */
+/* ====================================================================================================================================================== */
 
-// Function to get all pool data:
-const getPoolData = async () => {
+// Function to fetch all data:
+const fetch = async () => {
 
   // Initializations:
   let data = [];
+  let apiData = [];
   let startBatch = 0;
   let endBatch = batchSize;
 
-  // Fetching All SnowGlobes:
-  console.info(`\n  > Starting script calls...`);
+  // Fetching API Data:
+  apiData.push(...(await fetchAPI()));
+
+  // Fetching SnowGlobes:
   let globes = await fetchGlobes();
   maxProgress = globes.length;
 
-  // Fetching SnowGlobe Data:
+  // Fetching Data:
   while(progress < maxProgress) {
-    let promises = globes.slice(startBatch, endBatch).map(globe => (async () => {
-      if(!ignoreAddresses.includes(globe.toLowerCase())) {
-        let gauge = await fetchGauge(globe);
-        if(gauge != zero) {
-          let controller = await fetchController(globe);
-          let token = await fetchToken(globe);
-          let strategy = await fetchStrategy(controller, token.address);
-          let platform = await fetchPlatform(token, strategy, globe);
-          if(lpSymbols.includes(token.symbol)) {
-            let underlyingTokens = await fetchUnderlyingTokens(token.address);
-            data.push({platform, globe, strategy, gauge, controller, token, underlyingTokens});
-          } else {
-            data.push({platform, globe, strategy, gauge, controller, token});
-          }
-        }
-      }
-      updateProgress();
-    })());
-    await Promise.all(promises);
+    data.push(...(await fetchBatch(globes.slice(startBatch, endBatch), apiData)));
     startBatch += batchSize;
     endBatch += batchSize;
   }
 
-  // Writing JSON:
-  fs.writeFile('./scripts/poolsData.json', JSON.stringify(data, null, ' '), 'utf8', (err) => {
-    if(err) {
-      console.error(err);
+  // Sorting Data:
+  data.sort((a, b) => {
+    if(a.type === 'lp') {
+      if(b.type === 'lp') {
+        let nameSort = a.name.localeCompare(b.name);
+        if(nameSort === 0) {
+          return a.globe.localeCompare(b.globe);
+        } else {
+          return nameSort;
+        }
+      } else {
+        return -1;
+      }
+    } else if(b.type === 'lp') {
+      return 1;
     } else {
-      console.info(`  > Successfully updated JSON file.`);
+      let nameSort = a.name.localeCompare(b.name);
+      if(nameSort === 0) {
+        return a.globe.localeCompare(b.globe);
+      } else {
+        return nameSort;
+      }
     }
   });
-}
 
+  // JSON Output:
+  writeJSON(data, 'pools');
```

Comment:
for organization, create and write these outputs to a sub directory
- `pool_results/pools` 
- `pool_results/error-pools`

Reply:
Changed in f9530ff